src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -470,6 +472,11 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 			this.badgeClassForm.controls.competencies.addFromTemplate();
 		}
 		this.badgeCategory = currentBadgeCategory;
+
+		// fix the issue of missing badge-frame when changing type for first time (only with safari browser)
```

Comment:
This comment explains *what* this code is supposed to do, but I would like to understand *how* this code solves the issue. Maybe a comment explaining this would also be nice?

Reply:
Done.